Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/Batching/Batching.xaml.cs

```diff
@@ -0,0 +1,240 @@
+        {
+            string inputName = session.Model.InputFeatures[0].Name;
+            float totalDuration = 0;
+            for (int j = 0; j < input.Count; j++)
+            {
+                if (navigatingAwayFromPage)
+                    break;
+                var binding = new LearningModelBinding(session);
+                var start = HighResolutionClock.UtcNow();
+                binding.Bind(inputName, input[j]);
+                session.Evaluate(binding, "");
+                var stop = HighResolutionClock.UtcNow();
+                var duration = HighResolutionClock.DurationInMs(start, stop);
+                totalDuration += duration;
+            }
+            return totalDuration;
+        }
+
+        async private System.Threading.Tasks.Task ClassifyBatched(List<VideoFrame> inputImages, int batchSize)
+        {
+            float totalEvalDurations = 0;
+            for (int i = 0; i < 100; i++)
+            {
+                if (navigatingAwayFromPage)
+                    break;
+                UpdateEvalProgressUI(i);
+                float evalDuration = await System.Threading.Tasks.Task.Run(() => EvaluateBatched(BatchingSession_, inputImages, batchSize));
+                totalEvalDurations += evalDuration;
+            }
+            avgBatchDuration = totalEvalDurations / 100;
+        }
+
+        private static float EvaluateBatched(LearningModelSession session, List<VideoFrame> input, int batchSize)
+        {
+            // Evaluate
+            int numBatches = input.Count / batchSize;
+            string inputName = session.Model.InputFeatures[0].Name;
+            float totalDuration = 0;
+            for (int i = 0; i < numBatches; i++)
+            {
+                if (navigatingAwayFromPage)
+                    break;
+                var binding = new LearningModelBinding(session);
+                List<VideoFrame> batch = input.GetRange(batchSize * i, batchSize);
+                var start = HighResolutionClock.UtcNow();
+                binding.Bind(inputName, batch);
+                session.Evaluate(binding, "");
+                var stop = HighResolutionClock.UtcNow();
+                var duration = HighResolutionClock.DurationInMs(start, stop);
+                totalDuration += duration;
+            }
+            return totalDuration;
+        }
+
+        private int GetBatchSizeFromBatchSizeComboBox()
+        {
+            int batchSize;
+            if (BatchSizeComboBox.SelectedIndex == 0)
+                batchSize = 5;
+            else if (BatchSizeComboBox.SelectedIndex == 1)
+                batchSize = 10;
+            else if (BatchSizeComboBox.SelectedIndex == 2)
+                batchSize = 25;
+            else
+                batchSize = 50;
+            return batchSize;
+        }
+
+        private void UpdateEvalProgressUI(int attemptNumber)
+        {
+            EvalProgressText.Text = "Attempt " + attemptNumber.ToString() + "/100";
```

Comment:
can you use a const variable for the 100? better yet, use the same variable for the loop that does the evaluate, so if we ever change from doing 100 to another number, you just change it in one spot

Reply:
Done.